src/legacy/core_plugins/kibana/public/discover/np_ready/angular/context.js

```diff
@@ -68,53 +66,73 @@ getAngularModule().config($routeProvider => {
     });
 });
 
-function ContextAppRouteController(
-  $routeParams,
-  $scope,
-  AppState,
-  config,
-  $route,
-  getAppState,
-  globalState
-) {
+function ContextAppRouteController($routeParams, $scope, config, $route) {
   const filterManager = getServices().filterManager;
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
   const indexPattern = $route.current.locals.indexPattern.ip;
+  const { start, stop, appState, globalState, getAppFilters, getGlobalFilters } = getState(
+    config.get('context:defaultSize'),
+    indexPattern.timeFieldName,
+    config.get('state:storeInSessionStorage')
+  );
+  this.state = _.cloneDeep(appState.getState());
+  this.filters = _.cloneDeep([...getGlobalFilters(), ...getAppFilters()]);
+  filterManager.addFilters(this.filters);
+  start();
 
-  this.state = new AppState(createDefaultAppState(config, indexPattern));
-  this.state.save(true);
+  // take care of changes in State/URL
+  appState.subscribe(newState => {
+    if (newState && !_.isEqual(this.state, newState)) {
+      this.state = newState;
```

Comment:
Why don't you deep clone the `newState` here, but did it after initialization?

Reply:
@Dosant seems `getState` returns an object with readonly props but `subscribe` doesn't?

Reply:
Hm, is it? Looked through state container code, it seems like object should be properly frozen, no matter if it is coming from `.getState` or `subscribe`. 



Please note - readonly type was removed in scope of - https://github.com/elastic/kibana/pull/56571
We removed it because it was an overhead to always unbox to regular type. 
Even though type is removed, the state is still frozen in dev mode in runtime.